packages/r/re2/xmake.lua

```diff
@@ -14,26 +14,32 @@ package("re2")
     add_versions("2024.03.01", "7b2b3aa8241eac25f674e5b5b2e23d4ac4f0a8891418a2661869f736f03f57f4")
     add_versions("2024.04.01", "3f6690c3393a613c3a0b566309cf04dc381d61470079b653afc47c67fb898198")
     add_versions("2024.06.01", "7326c74cddaa90b12090fcfc915fe7b4655723893c960ee3c2c66e85c5504b6c")
+    add_versions("2024.07.02", "eb2df807c781601c14a260a507a5bb4509be1ee626024cb45acbd57cb9d4032b")
 
     add_deps("cmake", "abseil")
 
-    if is_plat("linux") then
+    if is_plat("linux", "bsd") then
         add_syslinks("pthread")
     end
 
     on_load(function (package)
         local version = package:version()
-        if version:eq("2024.06.01") and package:is_plat("mingw") then
+        if version:ge("2024.06.01") and package:is_plat("mingw") then
             package:add("syslinks", "Dbghelp")
         end
     end)
 
-    on_install("macosx", "linux", "windows", "mingw", "cross", function (package)
-        local configs = {"-DRE2_BUILD_TESTING=OFF"}
-        table.insert(configs, "-DCMAKE_CXX_STANDARD=17")
+    on_install(function (package)
+        local configs = {
+            "-DRE2_BUILD_TESTING=OFF",
+            "-DCMAKE_CXX_STANDARD=17",
+            "-Dabsl_DIR=" .. (package:dep("abseil"):installdir("lib", "cmake", "absl")),
```

Comment:
it's installdir is an empty directory if abseil is system library.

we should use `fetch` to get linkdirs.

```lua
local absldir
local abseil = package:dep("abseil"):fetch()
if abseil then
    for _, linkdir in ipairs(abseil.linkdirs) do
        local dir = path.join(linkdir, "cmake", "absl")
        if os.isdir(dir) then
                absldir = dir
                break
        end
    end
end
```